src/app/qgisapp.cpp

```diff
@@ -10837,6 +10838,41 @@ void QgisApp::projectCrsChanged()
   }
 }
 
+void QgisApp::projectTemporalRangeChanged()
+{
+  QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
+  QgsMapLayer *currentLayer = nullptr;
+
+  for ( QMap<QString, QgsMapLayer *>::const_iterator it = layers.constBegin(); it != layers.constEnd(); ++it )
+  {
+    currentLayer = it.value();
+
+    if ( currentLayer->dataProvider() )
+    {
+      QgsDataSourceUri uri;
+      uri.setEncodedUri( currentLayer->dataProvider()->dataSourceUri() );
+
+      if ( uri.hasParam( QStringLiteral( "temporalSource" ) ) &&
+           uri.param( QStringLiteral( "temporalSource" ) ) == QStringLiteral( "project" ) )
+      {
+        QgsDateTimeRange range = QgsProject::instance()->timeSettings()->temporalRange();
+        if ( range.begin().isValid() && range.end().isValid() )
+        {
+          QString time = range.begin().toString( Qt::ISODateWithMs ) + "/" +
+                         range.end().toString( Qt::ISODateWithMs );
+
+          uri.removeParam( QStringLiteral( "time" ) );
+          uri.setParam( QStringLiteral( "time" ), time );
+
+          currentLayer->dataProvider()->setDataSourceUri( uri.encodedUri() );
```

Comment:
Please do not use encodedUri!
... and keep in mind aware that WMST is not the only provider with temporal capabilities anymore.